### PR TITLE
Working on design system / Transfers

### PR DESCRIPTION
**Pull request by ThibaudDauce**

### CODE REVIEW

File: components/Datasets/DatasetsMetrics.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div
-    class="bg-white border rounded-xxs fr-mb-3w"
+    class="bg-white border rounded-xxs"
```
Comment (nicolaskempf57) [inline components/Datasets/DatasetsMetrics.vue:3]:
```suggestion
    class="bg-white border rounded"
```

File: components/BrandedButton/BrandedButton.vue [hunk 2]
```diff
@@ -1,75 +1,93 @@
...
+    :class="[colors, sizes, isDisabled ? '!opacity-50' : '']"
+    :disabled="isDisabled"
+    aria-disabled="isDisabled"
+    :role="as === 'a' ? 'link' : ''"
+    :href="isDisabled ? undefined : href"
   >
     <AdminLoader
       v-if="loading"
+      size="16"
       class="text-sm mr-1"
```
[the remark this answers is not in the document]
Comment (nicolaskempf57) [reply components/BrandedButton/BrandedButton.vue:14]: Also, the `mr-1` should be added only when `hasText` is true

File: components/BrandedButton/BrandedButton.vue [hunk 3]
```diff
@@ -1,75 +1,93 @@
...
+      :color="color === 'primary' ? 'white' : 'primary'"
     />
     <component
       :is="icon"
       v-if="icon"
```
Comment (nicolaskempf57) [inline components/BrandedButton/BrandedButton.vue:19]: What do you think of using a `v-else-if` here ? That way the button is not growing when loading

File: components/DesignSystem/BannerNotif.vue [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    </BannerNotif>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { Dataset, User } from '@datagouv/components'
+import { RiCheckLine, RiCloseLine, RiSendPlaneLine } from '@remixicon/vue'
+
+const { data: user } = await useAPI<User>(`/api/1/users/antonin-garrone`)
+const { data: dataset } = await useAPI<Dataset>(`/api/1/datasets/election-presidentielle-des-10-et-24-avril-2022-resultats-definitifs-du-1er-tour`)
```
Comment (nicolaskempf57) [inline components/DesignSystem/BannerNotif.vue:60]:
```suggestion
const { data: dataset } = await useAPI<Dataset>(`/api/1/datasets/base-sirene-des-entreprises-et-de-leurs-etablissements-siren-siret`)
```

Using an older dataset makes it work on old environments

File: components/TransferRequest.vue [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        <template v-if="transfer.recipient.class === 'User'">
+          {{ $t('to') }}
+          <a
+            :href="transfer.recipient.page"
+            class="link"
+          >{{ transfer.recipient.first_name }} {{ transfer.recipient.last_name }}</a>
+        </template>
+      </template>
+
+      <template #body>
```
Comment (nicolaskempf57) [inline components/TransferRequest.vue:52]: The body is missing the light grey left border

File: components/TransferBanner.vue [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+const loading = ref(false)
+const to = ref<Owned | null>(null)
+const comment = ref('')
+const existingTransfers = ref<Array<TransferRequest> | null>(null)
+
+const fetchTransfer = async () => {
+  if (existingTransfers.value) return
+
+  try {
+    existingTransfers.value = await $api<Array<TransferRequest>>('/api/1/transfer', {
```
Comment (nicolaskempf57) [inline components/TransferBanner.vue:104]:
```suggestion
    existingTransfers.value = await $api<Array<TransferRequest>>('/api/1/transfer/', {
```

File: components/TransferBanner.vue [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+    if (!to.value) return
+    let recipient
+    if (to.value.organization) {
+      recipient = { class: 'Organization', id: to.value.organization.id }
+    }
+    else {
+      recipient = { class: 'User', id: to.value.owner.id }
+    }
+
+    const transfer = await $api<TransferRequest>('/api/1/transfer', {
```
Comment (nicolaskempf57) [inline components/TransferBanner.vue:127]:
```suggestion
    const transfer = await $api<TransferRequest>('/api/1/transfer/', {
```

File: components/TransferBanner.vue [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+          />
+        </div>
+        <template #footer>
+          <div
+            v-if="existingTransfers && !existingTransfers.length"
+            class="flex-1 fr-btns-group fr-btns-group--right fr-btns-group--inline-reverse fr-btns-group--inline-lg fr-btns-group--icon-left"
+          >
+            <BrandedButton
+              color="primary"
+              :disabled="loading"
```
Comment (nicolaskempf57) [inline components/TransferBanner.vue:68]:
```suggestion
              :loading
```

**Review by nicolaskempf57 — COMMENTED** [review]
Some minor comments but otherwise it's really great !

File: components/BrandedButton/BrandedButton.vue (hunk 3 above)
Comment (ThibaudDauce) [reply components/BrandedButton/BrandedButton.vue]: Done in 57523ebea2949a6a1b8b4fa0355aa5dc4b899ece

File: components/BrandedButton/BrandedButton.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (ThibaudDauce) [reply components/BrandedButton/BrandedButton.vue]: Done in 57523ebea2949a6a1b8b4fa0355aa5dc4b899ece

File: components/TransferRequest.vue (hunk 5 above)
Comment (ThibaudDauce) [reply components/TransferRequest.vue:52]: Good catch! Fixed in 82b40d8

**Review by nicolaskempf57 — APPROVED** [review]
👏🏻

**Pull request merged**